src/server/ServerApp.tsx

```diff
@@ -11,19 +11,36 @@ export interface StaticContext {
   url?: string;
 }
 
+export type ReqContextType = {
+  host: string;
+  url: string;
+};
+
 interface Props {
   client: ApolloClient<Record<string, unknown>>;
-  context: StaticContext;
-  url: string;
+  staticContext: StaticContext;
+  reqContext: ReqContextType;
   i18n: i18nType;
 }
 
-const ServerApp: React.FC<Props> = ({ client, context, url, i18n }) => {
+export const ReqContext = React.createContext<ReqContextType>({
```

Comment:
useURIComponents.ts is now under hooks folder, but I can move context initialization to own file under contexts folder, if you meant that.

Reply:
I think it could be nice if the hook would also be in the same file with the context initialization.

Reply:
There is then two possibilites, move hook from hooks folder to context folder or move context from context folder to hooks folder :)
Hmm where is my push, it seems that because of slow pre-commit I keep forgetting to push :)

Reply:
now pushed, I thought I did it like one or two days ago